File: 01. Graph/01. Representataion/matrix.py
def create_matrix_graph(n, edges, is_undirected):
    graph = [[0]*n for _ in range(n)]

    for u, v in edges:
        graph[u][v] = 1

        if is_undirected:
            graph[v][u] = 1

    return graph
    

n = 6
edges = [[0,3], [1,2], [1,5], [2,4], [3,5], [5,4], [5,0]]

graph = create_matrix_graph(n, edges, False)
print(graph)

graph_undirected = create_matrix_graph(n, edges, True)
print(graph_undirected)

File: 01. Graph/01. Representataion/Matrix.java
class Matrix{
    public static void main(String[] args){
        boolean isUndirected = true;
        int numVertices = 5;
        boolean[][] graph = new boolean[numVertices][numVertices];

        int[][] edges = {
            {0, 1}, {0, 4},
            {1, 2}, {1, 3}, {1, 4},
            {2, 3},
            {3, 4}
            // Add more edges as needed
        };

        for (int[] edge : edges){
            int v = edge[0];
            int u = edge[1];

            graph[u][v] = true;
            // for undirected graph
            if (isUndirected)
                graph[v][u] = true;

        }

        // print the materix
        // Print the adjacency matrix
        for (int i = 0; i < numVertices; i++) {
            for (int j = 0; j < numVertices; j++) {
                System.out.print((graph[i][j] ? 1 : 0) + " ");
            }
            System.out.println();
        }
    }
}